File: 14-LongestCommonPrefix/14-LongestCommonPrefix.py
# Last updated: 07/04/2026, 11:43:33
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        udy=""
        strs = sorted(strs)
        small = strs[0]
        big = strs[-1]

        n = min(len(small),len(big))
        i  = 0 
        while i < n :
            if small[i] != big[i] :
                return udy
            udy+=small[i]
            i +=1

        return udy
    
             

File: 14-LongestCommonPrefix/14-LongestCommonPrefix.cpp
// Last updated: 07/04/2026, 11:43:25
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        
        sort(strs.begin(),strs.end());
        string s1 = strs.front();
        string s2 = strs.back();

        int n = min(s1.size(),s2.size());

        int i = 0;
        string ans;
        
        while(i<n){
            if ( s1[i] != s2[i]){
                break;
            }
            ans+=s1[i];
            i++;
        }
        return ans;


    }
};